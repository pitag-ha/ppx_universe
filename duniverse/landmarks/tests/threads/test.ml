let[@landmark] sleep _ =
  Unix.sleepf 0.1

let[@landmark] hello () =
  print_endline "Starting 100 threads ..."

let () =
  ((begin
      hello ();
      let a = Array.init 100 (Thread.create sleep) in
      Array.iter Thread.join a;
    end)[@landmark "main"]);
  let module L = Landmark_threads in
  let open L.Graph in
  let cg = L.export () in
  let agg = aggregate_landmarks cg in
  let all_nodes = nodes agg in
  print_endline "\nLandmark reached:";
  all_nodes
  |> List.map (fun {name; _} -> name)
  |> List.sort compare
  |> List.iter print_endline

