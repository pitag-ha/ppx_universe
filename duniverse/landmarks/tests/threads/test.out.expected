Starting 100 threads ...

Landmark reached:
ROOT
Test.hello
Test.main
